File: androidutil/src/test/java/com/akingyin/base/taskmanager/TestTask.java
package com.akingyin.base.taskmanager;

import com.akingyin.base.taskmanager.enums.TaskStatusEnum;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.Semaphore;

/**
 * @author king
 * @version V1.0
 * @ Description:
 * @ Date 2019/12/20 11:25
 */
public class TestTask  extends   AbsTaskRunner {

  private    int     index;

  private Semaphore   mSemaphore;

  private CountDownLatch  mCountDownLatch;

  public TestTask(int index, Semaphore semaphore, CountDownLatch countDownLatch) {
    this.index = index;
    mSemaphore = semaphore;
    mCountDownLatch = countDownLatch;
  }

  @Override public TaskStatusEnum onBefore() {
    System.out.println("onBefore="+index);

    return TaskStatusEnum.SUCCESS;
  }

  @Override public void onToDo() {
    System.out.println("onTodo");
    try {
      Thread.sleep(10);
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
    System.out.println("--------onToDo-----"+index);

    TaskOnSuccess();
  }
}

File: app/src/main/java/com/zlcdgroup/mrsei/task/TestTask.kt
package com.zlcdgroup.mrsei.task

import com.akingyin.base.taskmanager.AbsTaskRunner
import com.akingyin.base.taskmanager.enums.TaskStatusEnum

/**
 * @ Description:
 * @author king
 * @ Date 2019/12/20 11:43
 * @version V1.0
 */
class TestTask(var tagInfo:String) : AbsTaskRunner() {




    override fun onBefore(): TaskStatusEnum {
        println("onBefore=${tagInfo}")
        return  TaskStatusEnum.SUCCESS
    }

    override fun onToDo() {
        println("start to Do $tagInfo")
        Thread.sleep(10)
        println("end to Do $tagInfo")
        TaskOnSuccess()
    }
}